catalog/nsili/catalog-nsili-endpoint/src/main/java/org/codice/alliance/nsili/endpoint/requests/OrderRequestImpl.java

```diff
@@ -301,95 +302,126 @@ private boolean orderContainsSupportedDelivery() {
                 case FILESCOMPRESS: {
                     int currNum = 1;
                     for (ResourceContainer file : files) {
-                        ByteSource contents = getZip(file.getInputStream(), file.getName());
-                        String currNumPortion = String.format(FILE_COUNT_FORMAT, currNum);
-                        String currFileName =
-                                filename + "." + currNumPortion + "." + totalNumPortion
-                                        + packagingSpecFormatType.getExtension();
-                        writeFile(destination,
-                                contents.openStream(),
-                                contents.size(),
-                                currFileName,
-                                packagingSpecFormatType.getContentType());
-                        sentFiles.add(currFileName);
-                        currNum++;
+                        try (TemporaryFileBackedOutputStream fos = new TemporaryFileBackedOutputStream(
+                                MAX_MEMORY_SIZE);
+                                ZipOutputStream zipOut = new ZipOutputStream(fos)) {
+                            getZip(zipOut, file.getInputStream(), file.getName());
+                            ByteSource contents = fos.asByteSource();
+                            String currNumPortion = String.format(FILE_COUNT_FORMAT, currNum);
+                            String currFileName =
+                                    filename + "." + currNumPortion + "." + totalNumPortion
+                                            + packagingSpecFormatType.getExtension();
+                            writeFile(destination,
+                                    contents.openStream(),
+                                    contents.size(),
+                                    currFileName,
+                                    packagingSpecFormatType.getContentType());
+                            sentFiles.add(currFileName);
+                            currNum++;
+                        }
                     }
                 }
                 break;
                 case FILESZIP: {
-                    ByteSource zip = getZip(files);
-                    String currFileName = filename + packagingSpecFormatType.getExtension();
-                    writeFile(destination,
-                            zip.openStream(),
-                            zip.size(),
-                            currFileName,
-                            packagingSpecFormatType.getContentType());
-                    sentFiles.add(currFileName);
+                    try (TemporaryFileBackedOutputStream fos = new TemporaryFileBackedOutputStream(
+                            MAX_MEMORY_SIZE); ZipOutputStream zipOut = new ZipOutputStream(fos)) {
```

Comment:
@bdeining your example isn't the same thing, it only has one resource in the try clause. Mine has two resources in the try clause.